skyportal/models/photometry.py

```diff
@@ -248,3 +263,720 @@ def snr(self):
         unique=True,
     ),
 )
+
+
+class PhotometricSeries(conesearch_alchemy.Point, Base):
+    """
+    A series of photometric measurements taken
+    of the same object with the same telescope and filter,
+    continuously from mjd_start to mjd_end.
+
+    To initialize this function user must provide:
+    - data: an xarray dataset.
+            The data must contain an "mjds" coordinate,
+            and either a "fluxes" or a "mags" dataset.
+            Other datasets can include "fluxerr" or "magerr",
+            and any other auxiliary measurements that are stored
+            but generally not used by Skyportal like raw counts,
+            backgrounds, fluxes in other apertures, ra/dec, etc.
+            All the datasets must have the same length along
+            the mjd dimension.
+            Auxiliary data (not mags/fluxes) may have additional
+            dimensions that are saved but do not affect
+            calculations/plotting in Skyportal.
+      - series_id: a unique identifier of the set of images from
+            which this series was extracted. This is useful
+            for finding other objects that were observed
+            at the same time as this series.
+            Must be a string with only alphanumeric
+            characters, underscores, plus/minus signs,
+            and slashes (which are respected when
+            using the series_id to build the folder tree).
+    - series_obj_id: a unique index or name of the object inside
+            this observation series. Can be an index of the
+            star in the images, or an official designation.
+            Does not have to be the same as the obj_id.
+            E.g., the same physical object can have
+            a different identifier in different series.
+            Must be an integer or string with only
+            alphanumeric characters, underscores and "+-".
+            The total length of the folder tree,
+            including the root folder, series_id
+            and series_obj_id must be less than
+            255 characters.
+    """
+
+    __tablename__ = 'photometric_series'
+
+    def __init__(self, data, series_id, series_obj_id):
+        self.series_identifier = str(series_id)
+        self.series_obj_id = str(series_obj_id)
+
+        if len(data) == 0:
+            raise ValueError('Must supply a non-empty data structure.')
+
+        # verify data has all required fields
+        if 'fluxes' not in data and 'mags' not in data or 'mjds' not in data:
```

Comment:
makes it seem like `data` has to be dictionary-like with keys and values. Maybe also test for the instance types you allow for data